src/inmanta/execute/scheduler.py

```diff
@@ -430,3 +470,220 @@ def run(self, compiler: "Compiler", statements: Sequence["Statement"], blocks: S
             raise RuntimeException(stmt.expression, "not all statements executed %s" % all_statements)
 
         return True
+
+
+@dataclass(frozen=True)
+class EntityRelationship:
```

Comment:
Why have this instead of using the actual 'RelationAttribute' ?

I expect this object will be created a gazillion times?

Reply:
I also think this can be removed, if '_get_valid_type_hints' place the actual relation in the type hint